pkg/azure/provider/helpers/driver.go

```diff
@@ -115,11 +116,23 @@ func GetDiskNames(providerSpec api.AzureProviderSpec, vmName string) []string {
 	dataDisks := providerSpec.Properties.StorageProfile.DataDisks
 	diskNames := make([]string, 0, len(dataDisks)+1)
 	diskNames = append(diskNames, utils.CreateOSDiskName(vmName))
-	if !utils.IsSliceNilOrEmpty(dataDisks) {
-		for _, disk := range dataDisks {
-			diskName := utils.CreateDataDiskName(vmName, disk)
-			diskNames = append(diskNames, diskName)
-		}
+	dataDiskNames := getDataDiskNames(providerSpec, vmName)
+	if dataDiskNames != nil {
+		diskNames = append(diskNames, dataDiskNames...)
+	}
+	return diskNames
+}
+
+// getDataDiskNames creates disk names for all configured DataDisks in the provider spec.
+func getDataDiskNames(providerSpec api.AzureProviderSpec, vmName string) []string {
+	dataDisks := providerSpec.Properties.StorageProfile.DataDisks
+	if utils.IsSliceNilOrEmpty(dataDisks) {
+		return nil
```

Comment:
There is no reason to return nil. Create a variable for the slice and return that slice. So the return value will never be nil but either an empty or non-zero length slice. This way the caller also does not need to specifically handle nil values. They can always just do an append.